src/libraries/Microsoft.Extensions.DependencyModel/src/LibraryCollectionExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Extensions.DependencyModel
+{
+    using System;
+    using System.Collections.Generic;
+
+    internal static class LibraryCollectionExtensions
+    {
+        public static Dictionary<string, T> LibraryCollectionToDictionary<T>(this IReadOnlyList<T> collection) where T : Library
```

Comment:
This should only be necessary when building for .NET Framework. On .NET Core, `Dictionary.Add` already adds the duplicate key value to the exception message.

https://github.com/dotnet/runtime/blob/3d5d26181cd7bc07ea6c6f87710c14ccd043b415/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs#L546-L548

You can use `#if NETFRAMEWORK` for the work-around code to check for duplicates manually. And when not on .NET Framework, just straight call `.ToDictionary` as before.